pkg/encryption/crypter.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/binary"
+	"io"
+	"time"
+
+	"github.com/pingcap/kvproto/pkg/encryptionpb"
+	"github.com/pkg/errors"
+)
+
+const (
+	ivLengthCTR = 16
+	ivLengthGCM = 12
+)
+
+// CheckEncryptionMethodSupported check whether the encryption method is currently supported.
+// This is to handle future extension to encryption methods on kvproto side.
+func CheckEncryptionMethodSupported(method encryptionpb.EncryptionMethod) error {
+	switch method {
+	case encryptionpb.EncryptionMethod_AES128_CTR:
+		return nil
+	case encryptionpb.EncryptionMethod_AES192_CTR:
+		return nil
+	case encryptionpb.EncryptionMethod_AES256_CTR:
+		return nil
+	default:
+		name, ok := encryptionpb.EncryptionMethod_name[int32(method)]
+		if ok {
+			return errors.Errorf("invalid encryption method %s", name)
+		}
+		return errors.Errorf("invalid encryption method %d", int32(method))
+	}
+}
+
+// KeyLength return the encryption key lenght for supported encryption methods.
+func KeyLength(method encryptionpb.EncryptionMethod) int {
+	switch method {
+	case encryptionpb.EncryptionMethod_AES128_CTR:
+		return 16
+	case encryptionpb.EncryptionMethod_AES192_CTR:
+		return 24
+	case encryptionpb.EncryptionMethod_AES256_CTR:
+		return 32
+	default:
+		panic("unsupported encryption method")
```

Comment:
I tend to throw an error here. Why use panic ?

Reply:
I think it's better to use `log.Fatal`